draft-ietf-quic-tls.md

```diff
@@ -1086,13 +1089,26 @@ before the final TLS handshake messages are received.  A client will be unable
 to decrypt 1-RTT packets from the server, whereas a server will be able to
 decrypt 1-RTT packets from the client.
 
-However, a server MUST NOT process data from incoming 1-RTT protected packets
-before verifying either the client Finished message or - in the case that the
-server has chosen to use a pre-shared key - the pre-shared key binder (see
-Section 4.2.11 of {{!TLS13}}).  Verifying these values provides the server with
-an assurance that the ClientHello has not been modified.  Packets protected with
-1-RTT keys MAY be stored and later decrypted and used once the handshake is
-complete.
+Until the handshake is complete, the connection and key exchange are not
+properly authenticated at the server.  In the case that the server has chosen
+to use a pre-shared key, validating the client's pre-shared key binder (see
+Section 4.2.11 of [TLS13]) provides assurances on the identity of the client
+but does not guarantee that previous handshake messages have not been tampered
+with.  Even though 1-RTT keys are available to a server after receiving the
```

Comment:
This is hard to read. Do you mean CH1/HRR? Because if so, that's not correct, because CH1 and HRR are part of the CH2 binder.

Reply:
Can you suggest better text?

Reply:
I don't know what it is trying to say.

Reply:
Clarification question: am I correct in assuming that the rewrite of this paragraph is editorial? I'm asking this because, assuming that is the case, all the unresolved discussions on the PR seem to be editorial.

Reply:
@kazuho I do agree that this is editorial. This text tries to explain the rationale for why the server MUST NOT process 1-RTT frames until it's completed the handshake. If the text is unclear to @ekr then I agree that it needs to be better written. Can someone offer text suggestions for wordsmithing here? The reason I had modified the text in the first place was that this seemed to flow better now that we've formalized the definition of handshake complete in its own subsection. But I'm happy to change this, or even happier to let the editors decide how best to word this.

Reply:
I think that there are two things to say, and you want to use two paragraphs (or more).

1. A server can send 1-RTT data to the client before completing the handshake, but it does so with limitations:

   a. The client is not authenticated unless there is a PSK binder.
   b. The client has not demonstrated liveness, so ...
   c. Any received 0-RTT data that the server responds to might be due to a replay attack.

2. Otherwise, until the handshake is complete, don't send or receive 1-RTT packets.  Endpoints will build strong assumptions about the identity of peers when they send or process 1-RTT packets that might be incorrect if those packets are sent without allowing the handshake to complete.

Reply:
Yes, this seems good.

Reply:
I've used these suggestions to refactor the text